### PR TITLE
[XLA] Make RemoveUnusedOperandFromSort a seperate pass

### PR DESCRIPTION
**Pull request by georgepaw**
Current implementation will cause an infinite loop if we put the AlgebraicSimplifier inside the HloPassFix.

### CODE REVIEW

**Review by sanjoy — CHANGES_REQUESTED** [review]
See @mkuperst 's reply on https://groups.google.com/d/msg/xla-dev/Qf-3dPULLEA/1r4QIPB_DwAJ

**Comment by mkuperst** [comment]
I was wrong on that thread, no objections to this PR.
(@majnemer also agrees, FWTW)

File: tensorflow/compiler/xla/service/sort_simplifier.cc
Comment (sanjoy) [inline tensorflow/compiler/xla/service/sort_simplifier.cc]: I know you're just moving code, but should be `graph has been modified`.

**Review by sanjoy — CHANGES_REQUESTED** [review]
I think you also need to add `SortSimplifier` to the optimization pipeline for the CPU and GPU compilers so that we don't regress performance.

File: tensorflow/compiler/xla/service/sort_simplifier.cc
Comment (georgepaw) [reply tensorflow/compiler/xla/service/sort_simplifier.cc]: Done

**Comment by georgepaw** [comment]
Yeah, I didn't add it as I just wanted to check this is the right thing to do. Let me know if the pass needs to be added in more places.

File: tensorflow/compiler/xla/service/sort_simplifier.h
Comment (sanjoy) [inline tensorflow/compiler/xla/service/sort_simplifier.h]: 2019

**Review by sanjoy — CHANGES_REQUESTED** [review]
LGTM modulo nit.

Note to Googlers -- please do not auto-merge this; I'll do it manually myself since it requires some matching changes to internal code.

File: tensorflow/compiler/xla/service/sort_simplifier.h
Comment (georgepaw) [reply tensorflow/compiler/xla/service/sort_simplifier.h]: @sanjoy should it be 2018 or 2017?

Comment (sanjoy) [reply tensorflow/compiler/xla/service/sort_simplifier.h]: Oops, made the comment on the wrong file; I meant on `sort_simplifier_test.cc`.

Comment (georgepaw) [reply tensorflow/compiler/xla/service/sort_simplifier.h]: Thanks, done now :)

**Review by sanjoy — DISMISSED** [review]
Approved, but please do not merge automatically.  I'll do it manually myself.

**Comment by hgadig** [comment]
@sanjoy  So, do you want me to proceed with running kokoro build checks ?

**Comment by sanjoy** [comment]
@hgadig Yes, Kokoro is fine; I just don't want this change to automatically get merged internally.

**Comment by georgepaw** [comment]
Fixed the nit in the `tensorflow/compiler/xla/service/BUILD`

**Comment by hgadig** [comment]
@sanjoy  In order to proceed with this PR, your approval is required. Please approve or suggest changes(if any)

**Review by sanjoy — APPROVED** [review]

**Pull request merged**
